### PR TITLE
fix: handle course access errors in Course Home side of things too

### PR DESCRIPTION
**Pull request by mikix**
The courseware was properly reading the access errors and redirecting the user as appropriate (like to the dashboard or whatever).

This requires a backend change to push the error along.

### CODE REVIEW

File: src/course-home/dates-tab/DatesTab.test.jsx [hunk 1]
```diff
@@ -295,4 +298,45 @@ describe('DatesTab', () => {
...
+      render(component);
+      await waitForElementToBeRemoved(screen.getByRole('status'));
+    }
+
+    beforeEach(() => {
+      axiosMock.onGet(datesUrl).reply(200, datesTabData);
+      history.push(`/course/${courseId}/dates`); // so tab can pull course id from url
+    });
+
+    it('should go to course survey for an survey_required error code', async () => {
```
Comment (ciduarte) [inline src/course-home/dates-tab/DatesTab.test.jsx]: nit: `an` should be `a`

Comment (ciduarte) [reply src/course-home/dates-tab/DatesTab.test.jsx]: Another nit, can we change `should go` in all of these tests to `redirects`? Feels a little more clear when scanning

File: src/courseware/CoursewareContainer.jsx [hunk 2]
```diff
@@ -338,10 +296,9 @@ const sectionShape = PropTypes.shape({
 });
 
 const courseShape = PropTypes.shape({
-  canLoadCourseware: PropTypes.shape({
-    errorCode: PropTypes.string,
-    additionalContextUserMessage: PropTypes.string,
-  }).isRequired,
+  celebrations: PropTypes.shape({
+    firstSection: PropTypes.bool,
+  }),
```
Comment (ciduarte) [inline src/courseware/CoursewareContainer.jsx:301]: Did you add this in because there was a props warning or something?

**Review by ciduarte — APPROVED** [review]
Just some nits and a question! LGTM

File: src/courseware/CoursewareContainer.jsx (hunk 2 above)
Comment (mikix) [reply src/courseware/CoursewareContainer.jsx:301]: I didn't notice one - just was updating the shape to be accurate to what we currently use in this file.

File: src/course-home/dates-tab/DatesTab.test.jsx (hunk 1 above)
Comment (mikix) [reply src/course-home/dates-tab/DatesTab.test.jsx]: :+1:

**Pull request merged**